File: src/main/java/com/floorestimatepro/model/Calibration.java
package com.floorestimatepro.model;

public record Calibration(double feetPerPixel){

    public static Calibration fromKnownLength( PlanPoint a, PlanPoint b, double knownFeet){

        double pixelDistance = a.distanceTo(b);
        if(pixelDistance == 0){
            throw new IllegalArgumentException("calibration points must not be the same");
        }

        if(knownFeet <= 0){
            throw new IllegalArgumentException("known length must be positive");
        }

        double scale  = knownFeet / pixelDistance;

        return new Calibration(scale);
    }

    public double toFeet(double pixelLength){
        return pixelLength * feetPerPixel;
    }

    public double toSquareFeet(double pixelArea){
        double s = this.feetPerPixel;
        return pixelArea * s * s;
    }


}

File: python/src/floor_estimate_pro/model/calibration.py
from __future__ import annotations

from dataclasses import dataclass

from floor_estimate_pro.model.plan_point import PlanPoint


@dataclass(frozen=True)
class Calibration:
    feet_per_pixel: float

    @staticmethod
    def from_known_length(a: PlanPoint, b: PlanPoint, known_feet: float) -> Calibration:
        pixel_distance = a.distance_to(b)
        if pixel_distance == 0:
            raise ValueError("calibration points must not be the same")
        if known_feet <= 0:
            raise ValueError("known length must be positive")
        scale = known_feet / pixel_distance
        return Calibration(scale)

    def to_feet(self, pixel_length: float) -> float:
        return pixel_length * self.feet_per_pixel

    def to_square_feet(self, pixel_area: float) -> float:
        s = self.feet_per_pixel
        return pixel_area * s * s
